src/openforms/formio/components/custom.py

```diff
@@ -106,12 +107,18 @@ class NPFamilyMembers(BasePlugin):
 
     @staticmethod
     def _get_handler() -> Callable[[str], list[tuple[str, str]]]:
-        handlers = {
-            FamilyMembersDataAPIChoices.haal_centraal: get_np_children_haal_centraal,
-            FamilyMembersDataAPIChoices.stuf_bg: get_np_children_stuf_bg,
-        }
-        config = FamilyMembersTypeConfig.get_solo()
-        return handlers[config.data_api]
+        match FamilyMembersTypeConfig.get_solo().data_api:
```

Comment:
This requires backporting to Open Forms 2.0.x which is on Python 3.8, for which this syntax won't work.

I think a more robust approach here would be to use StUF-BG as fallback if nothing is explicitly configured, as that is what the bulk of the organizations use (only a few large municipalities are early adopters of Haal Centraal).

So to address the key error: `handlers.get(config.data_api, get_np_children_stuf_bg)`

I don't see how any other values than one of the `FamilyMembersDataAPIChoices` and the empty value `""` would end up in the database. And to ensure that every choice is present in the mapping, I'd rather see a module-level developer helper-check along the lines of:

```python
for value in FamilyMembersDataAPIChoices.values:
    assert value in HANDLER_MAPPING
```

Personally I also don't find that the match statement helps with readability, a mapping lookup is much flatter and simpler to process IMO.